---
title: C26111 | Microsoft Docs
ms.custom: ''
ms.date: 11/04/2016
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-ide-code-analysis
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
caps.latest.revision: 8
author: mikeblome
ms.author: mblome
manager: ghogen
ms.workload:
- multiple
ms.openlocfilehash: dadc0c15e07595d3a35d4cb4a92f257bdcdf3490
ms.sourcegitcommit: 32f1a690fc445f9586d53698fc82c7debd784eeb
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 12/22/2017
---
# <a name="c26111"></a>C26111
Warnung C26111: wegen eines Fehlers beim Sperre Aufrufer \<Sperre > vor dem Aufruf von Funktion \<Func >.  
  
 Die Anmerkung `_Requires_lock_not_held_` erzwingt eine Voraussetzung, dass die Anzahl der Sperren für die angegebene Sperren größer als 0 (null) sein kann, wenn die Funktion aufgerufen wird. C26111 Warnung wird ausgegeben, wenn eine Funktion ein Fehler auftritt, um die Sperre aufzuheben, bevor sie eine andere Funktion aufruft.  
  
## <a name="example"></a>Beispiel  
 Im folgende Beispiel wird die Warnung C26111 generiert, da die `_Requires_lock_not_held_` Vorbedingung verletzt wird, durch den Aufruf von `DoNotLock` innerhalb des Abschnitts gesperrt.  
  
```  
  
typedef struct _DATA   
{  
    CRITICAL_SECTION cs;  
    int d;  
} DATA;  
  
_Requires_lock_not_held_(p->cs)   
  
void DoNotLock(DATA* p)  
{  
    EnterCriticalSection(&p->cs);   
    p->d = 0;  
    LeaveCriticalSection(&p->cs);  
}  
  
void LockedFunction(DATA* p)   
{   
    EnterCriticalSection(&p->cs);   
    DoNotLock(p); // Warning C26111  
    LeaveCriticalSection(&p->cs);  
}  
  
```